gratipay/models/package/__init__.py

```diff
@@ -40,3 +51,40 @@ def from_names(cls, package_manager, name):
         """
         return cls.db.one("SELECT packages.*::packages FROM packages "
                           "WHERE package_manager=%s and name=%s", (package_manager, name))
+
+
+    @property
+    def team(self):
+        """A computed attribute, the :py:class:`~gratipay.models.team.Team`
+        linked to this package if there is one, otherwise ``None``. Makes a
+        database call.
+        """
+        return self.load_team(self.db)
+
+
+    def load_team(self, cursor):
+        """Given a database cursor, return a
+        :py:class:`~gratipay.models.team.Team` if there is one linked to this
+        package, or ``None`` if not.
+        """
+        return cursor.one('SELECT t.*::teams FROM teams t WHERE t.id=%s', (self.team_id,))
+
+
+    def get_or_create_linked_team(self, cursor, owner):
+        """Given a db cursor and :py:class:`Participant`, return a
+        :py:class:`~gratipay.models.team.Team`.
+        """
+        team = self.load_team(cursor)
+        if not team:
+            slug = str(uuid.uuid4()).lower()
+            team = Team.insert( slug=slug
```

Comment:
TIL about `INSERT OR IGNORE`. Looks like that's a MySQL thing?

The [alternative](http://stackoverflow.com/questions/1009584/how-to-emulate-insert-ignore-and-on-duplicate-key-update-sql-merge-with-po) in Postgres is to try/except, but that [doesn't work](https://github.com/gratipay/postgres.py/issues/60) with cursors. I tried! :-)